src/Debugging/Debugger.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Debugging;
+
+use Saloon\Contracts\DebuggingDriver;
+use Saloon\Debugging\Drivers\RayDebugger;
+use Saloon\Exceptions\UnknownDriverException;
+use Saloon\Debugging\Drivers\ErrorLogDebugger;
+use Saloon\Debugging\Drivers\SystemLogDebugger;
+
+class Debugger
+{
+    /**
+     * Globally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected static array $globalRegisteredDrivers = [];
+
+    /**
+     * Locally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected array $registeredDrivers = [];
+
+    /**
+     * Drivers that have been subscribed to
+     *
+     * @var array<int, string>
+     */
+    protected array $useDrivers = [];
+
+    /**
+     * Denotes if we send the request to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showRequest = false;
+
+    /**
+     * Denotes if we send the response to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showResponse = false;
+
+    public function __construct()
+    {
+        $this->registerDriver(new RayDebugger);
+        $this->registerDriver(new ErrorLogDebugger);
+        $this->registerDriver(new SystemLogDebugger);
+    }
+
+    /**
+     * Register a driver globally
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return static
+     */
+    public static function registerGlobalDriver(DebuggingDriver $driver): static
+    {
+        static::$globalRegisteredDrivers[$driver->name()] = $driver;
+
+        return new static;
+    }
+
+    /**
+     * Register a driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return $this
+     */
+    public function registerDriver(DebuggingDriver $driver): static
+    {
+        $this->registeredDrivers[$driver->name()] = $driver;
+
+        return $this;
+    }
+
+    /**
+     * Subscribe to a given driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver|string $driver A DebuggingDriver or the name one of a registered one.
+     *
+     * @return $this
+     * @throws \Saloon\Exceptions\UnknownDriverException
+     */
+    public function usingDriver(DebuggingDriver|string $driver): static
+    {
+        if ($driver instanceof DebuggingDriver) {
+            $this->registerDriver($driver);
+        }
+
+        $driverName = is_string($driver) ? $driver : $driver->name();
+
+        // We'll validate that the driver exists
+
+        $registeredDriverNames = array_keys($this->getRegisteredDrivers());
+
+        if (! in_array($driverName, $registeredDriverNames, true)) {
+            throw new UnknownDriverException(sprintf('Unable to find the "%s" driver. Registered drivers: %s', $driverName, implode(', ', $registeredDriverNames)));
+        }
```

Comment:
I think this is technically a micro-optimisation, given the realistically low amount of items, but _I feel like_ repetitively sending requests, there's a bit of time to gain here.
Doing a hash lookup should be a lot quicker than `in_array()` the key.
We'd also be using language constructs, rather than function calls.

Throwing Exceptions in expressions has been supported since PHP 8.0.

```suggestion
        // We'll validate that the driver exists

        $registeredDrivers = $this->getRegisteredDrivers();
        $registeredDrivers[$driverName] ?? throw new UnknownDriverException(sprintf(
            'Unable to find the "%s" driver. Registered drivers: %s',
            $driverName,
            implode(', ', array_keys($registeredDrivers)),
        ));
```

Reply:
Side note:
Normally you'd store the result of
```php
$registeredDrivers[$driverName] ?? 'something';
```

But since we don't need the result, I opted to omit it.